bcda/bcdacli/cli.go

```diff
@@ -658,3 +680,74 @@ func setBlacklistState(cmsID string, blacklistState bool) error {
 
 	return db.Model(&aco).Update("blacklisted", blacklistState).Error
 }
+
+// CCLF file name pattern and regex
+const cclfPattern = `((?:T|P)\.BCD\..*\.ZC\d*)Y(\d{2}\.D\d{6}\.T\d{7})`
+
+var cclfregex = regexp.MustCompile(cclfPattern)
+
+func RenameCCLF(name string) string {
+	return cclfregex.ReplaceAllString(name, "${1}R${2}")
+}
+
+func CloneCCLFZips(path string) error {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+
+	// Iterate through all cclf zip files in provided directory
+	for _, f := range files {
+		// Make sure to not clone non CCLF files in case the wrong directory is given
+		if !cclfregex.MatchString(f.Name()) {
+			log.Infof("Skipping file `%s`, does not match expected name... ", f.Name())
+			continue
+		}
+		fn, err := CloneCCLFZip(path, f.Name())
+		if err != nil {
+			return err
+		}
+		log.Infof("Created runout file: %s", fn)
+	}
+	return nil
+}
+
+func CloneCCLFZip(path, name string) (string, error) {
+	// Open source zip file for cloning
+	zr, err := zip.OpenReader(filepath.Join(path, name))
+	if err != nil {
+		return "", err
+	}
+	defer zr.Close()
+
+	// Create destination runout zip file with proper nomenclature
+	filename := RenameCCLF(name)
+	newf, err := os.Create(filepath.Join(path, filename))
+	if err != nil {
+		return "", err
+	}
+	defer newf.Close()
```

Comment:
Didn't realize it was unsafe, but the gosec error seems valid and should be addressed. One approach would be to use named returns. So something like this:
```
func CloneCCLFZip(path, name string) (newFile string, err error)
```

and then this is updated to something like this
```
defer func() {
    if err1 := newf.Close(); err1 != nil && err == nil {
            err = err1
    }
}()
```

So we won't ignore the main error, but if we have an error that was caused by closing the file, we'll return it out.

Reply:
Looking more into this, there's a helper function that you can use `utils.CloseFileAndLogError`